### PR TITLE
Scala Native 0.5 Build

### PR DESCRIPTION
**Pull request by cheeseng**
This PR is based on https://github.com/scalatest/scalatest/pull/2318 , just minor tweaked for publishing 3.2.18 build for scala-native 0.5.

### CODE REVIEW

**Comment by catap** [comment]
@cheeseng this introduces update of dependencies and to be on safe side I suggest to release it as 3.2.19.

**Comment by cheeseng** [comment]
> @cheeseng this introduces update of dependencies and to be on safe side I suggest to release it as 3.2.19.

@catap I think it should be fine, there's a scala-native 0.4 already published previously for 3.2.18 and it still can be used, or user may choose to use the 0.5 build (staged) if want to.  Let's see if @bvenners likes it, starting 3.2.19 we should just have build for 0.5 though.  :)

**Comment by catap** [comment]
On Thu, 18 Apr 2024 17:10:30 +0200,
Chua Chee Seng wrote:
> 
> > @cheeseng this introduces update of dependencies and to be on safe side I suggest to release it as 3.2.19.
> 
> @catap I think it should be fine, there's a scala-native 0.4 already published previously for 3.2.18 and it still can be used, or user may choose to use the 0.5 build (staged) if want to.  Let's see if @bvenners likes it, starting 3.2.19 we should just have build for 0.5 though.  :)
> 

Keep in mind that here I've updated scala-xml for example, and to have
3.2.18 which depends on different version of scala-xml may lead to issues.

-- 
wbr, Kirill

**Comment by catap** [comment]
@cheeseng unfortently I was forced to upgrade scala-js to match version of sjs inside dependencies like scala-xml.

File: project/DottyBuild.scala [hunk 1]
```diff
@@ -289,7 +289,7 @@ trait DottyBuild { this: BuildCommons =>
       initialCommands in console := """|import org.scalatest._
                                        |import org.scalactic._
                                        |import Matchers._""".stripMargin,
-      libraryDependencies += "org.scala-lang.modules" %%% "scala-xml" % "2.3.0",
+      libraryDependencies += "org.scala-lang.modules" %%% "scala-xml" % "2.1.0",
```
Comment (catap) [inline project/DottyBuild.scala:292]: ah, this is dedicated project for Scala3Js. Got it.

**Review by bvenners — APPROVED** [review]

**Comment by cheeseng** [comment]
@catap Fyi Artifacts for scala-native 0.5 are up.

**Pull request merged**